### PR TITLE
Add msfiddle 2.1.0

### PR DESCRIPTION
**Pull request by RJMW**
New recipe for **FIDDLE** (Formula IDentification by Deep LEarning) — predicts molecular formulas from tandem mass spectra using Orbitrap and Q-TOF models. It can optionally incorporate BUDDY/msbuddy and SIRIUS results as external candidates for rescoring.

- Upstream: https://github.com/JosieHong/msfiddle
- Paper: [doi:10.1038/s41467-025-66060-9](https://doi.org/10.1038/s41467-025-66060-9)
- Licence: Apache-2.0

### Built and tested

Built in `quay.io/bioconda/bioconda-utils-build-env-cos7` under `--platform linux/amd64`:

```
BUILD START: ['msfiddle-2.1.0-py_0.conda']
TEST START / TEST END        exit 0
```

`bioconda-utils lint` (4.4.1) clean. Source sha256 verified against PyPI.

The test section exercises all three console scripts, and additionally asserts that the packaged `config/` directory and the bundled demo `.mgf` fixtures were actually installed — a `package_data` failure that a bare `import msfiddle` would not catch.

### Two dependency decisions worth a reviewer's eye

**`numpy >=1.20.0,<2.0a0`** — upstream declares `numpy>=1.20.0,<2.0.0`, and the upper bound is carried verbatim because it is a hard constraint rather than a courtesy pin.

**`pytorch` is declared explicitly** rather than relying on upstream's `[inference]` extra. Installing the extra through pip would hide the dependency from the conda solver; declaring it makes the environment resolvable and complete.

### Checkpoints

Pretrained checkpoints are **not bundled**, and nothing in the build or test section downloads them — `msfiddle-download-models` exists for that and is expected to be run ahead of time by whoever deploys the package. `msfiddle-checkpoint-paths` reports where they are looked for.

### CODE REVIEW

**Comment by RJMW** [comment]
@BiocondaBot please add label

File: recipes/msfiddle/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    - {{ pin_subpackage(name, max_pin="x") }}
+
+requirements:
+  host:
+    - python >=3.8
+    - pip
+    - setuptools
+  run:
+    - python >=3.8
+    # Upstream declares numpy>=1.20.0,<2.0.0. The upper bound is a hard
```
Comment (bgruening) [inline recipes/msfiddle/meta.yaml]: Please, clean this up :)

**Comment by RJMW** [comment]
@bgruening, I cleaned up the comments throughout the YAML as suggested. (sorry mixed it up with the other one). Will do the other one next.

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
Thanks!
